perfkitbenchmarker/linux_packages/blazemark.py

```diff
@@ -0,0 +1,206 @@
+# Copyright 2014 PerfKitBenchmarker Authors. All rights reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Module containing blazemark installation and cleanup functions."""
+
+import copy
+import logging
+import os
+
+from perfkitbenchmarker import regex_util
+from perfkitbenchmarker.linux_packages import blaze
+from perfkitbenchmarker.linux_packages import fortran
+
+BLAZEMARK_FOLDER = 'blazemark'
+BLAZEMARK_DIR = os.path.join(blaze.BLAZE_DIR, BLAZEMARK_FOLDER)
+CONFIG = 'Configfile_blazemark'
+
+THROUGHPUT_HEADER_REGEX = (
+    r'(\w+[\w\- ]+\w+)\s*(\([0-9]+% filled\))*\s*\[([\w/]+)\]:([0-9\s.]+)')
+THROUGHPUT_RESULT_REGEX = r'([0-9]+)\s*([0-9.]+)'
+FILLED_REGEX = r'([0-9.]+)% filled'
+
+LIBS = frozenset([
+    'C-like', 'Classic', 'Blaze', 'Boost uBLAS', 'Blitz++',
+    'GMM++', 'Armadillo', 'MTL', 'Eigen'])
+
+
+def GetBinaries(vm):
+  """Find available blazemark binaries."""
+  out, _ = vm.RemoteCommand('ls %s' % (
+      os.path.join(BLAZEMARK_DIR, 'bin')))
+  return out.split()
+
+
+def _SimplfyLibName(name):
+  """Simply library name parsed from output.
+
+  Args:
+    name: string. Name parsed from blazemark output.
+
+  Returns:
+    A simplified name defined in LIBS.
+  """
+  for lib in LIBS:
+    if lib in name:
+      return lib
+  return name
+
+
+def _ParseThroughput(out, test):
+  """Parse throughput section of blazemark results.
+
+  Args:
+    out: string. Blazemark output in raw string format.
+    test: string. Name of the test ran.
+
+  Returns:
+    A list of samples. Each sample if a 4-tuple of (benchmark_name, value, unit,
```

Comment:
why not return sample.Sample objects directly so you don't need to do the conversion later?